lib/Migration/Version23000Date2022022912000001.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * @copyright Copyright (c) 2021 Carl Schwan <carl@carlschwan.eu>
+ *
+ * @author Carl Schwan <carl@carlschwan.eu>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+declare(strict_types=1);
+
+namespace OCA\MonthlyStatusEmail\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\SimpleMigrationStep;
+use OCP\Migration\IOutput;
+use OCP\IDBConnection;
+
+class Version23000Date2022022912000001 extends SimpleMigrationStep {
+
+	/** @var IDBConnection */
+	protected $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = \OC::$server->get(IDBConnection::class);
+	}
+
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+		$query = $this->connection->getQueryBuilder();
+
+		$query->select($query->func()->count('*', 'num_entry'), 'user_id')
+			->from('notification_tracker');
+
+		$result = $query->executeQuery();
+		while ($row = $result->fetch()) {
+			$userId = $row['user_id'];
+			$entriesToDelete = (int)$row['num_entry'] - 1;
+			if ($entriesToDelete > 0) {
+				$delete = $this->connection->getQueryBuilder();
+				$delete->delete('notification_tracker')
+					->where($delete->expr()->eq('user_id', $delete->createNamedParameter($userId)))
+					->setMaxResults($entriesToDelete);
```

Comment:
setMaxResult on delete is only supported on mysql

Also you are missing the `execute` ?

Reply:
added the excute and the app is only running on a MySQL server for now so probably not a big issue...

Reply:
https://github.com/nextcloud/monthly_status_email/blob/master/appinfo/info.xml

Then please specify it as dependency to ensure this?
https://github.com/nextcloud/twofactor_webauthn/blob/v0.3.0/appinfo/info.xml#L21

or simply redo the query so it works on all dbs:
```php
$query->select($query->func()->max('id', 'newest_entry'), 'user_id')
```
And then delete based on this:
```php

				$delete->delete('notification_tracker')
					->where($delete->expr()->eq('user_id', $delete->createNamedParameter($row['user_id'])))
					->andWhere($delete->expr()->neq('id', $delete->createNamedParameter($row['newest_entry'])))
```